### ISSUE TITLE
Mimic db disappeared from my mac ...

### ISSUE THREAD

**Issue by GrigorijSchleifer** [MIT-LCP/mimic-code#802]
Hi everyone, 

suddenly I can't find or connect to the mimic database on my local machine (ProductName: Mac OS X
ProductVersion: 10.15.5, BuildVersion: 19F101).  I haven't used mimic on my Mac for a while and now I am not able to see the database at all. Postgresql is installed and I am able to connect to "default" and other db's. 

I think that mimic was installed in <strong>/usr/local/var/postgres</strong> and the directory info tells me that the folder is 63.246.509.246 bytes (63,98 GB on disk) with 3.453 items in it. I am guessing that mimic was installed there but don't know for sure. On a Mac machine that has no mimic or postgres installed, <strong>/usr/local/</strong> is empty. 

<strong>\l</strong> or <strong>\list</strong> return both all "visible" databases:

```code
                                          List of databases
 Name        |       Owner       | Encoding |   Collate   |    Ctype    |   Access privileges
-------------------+-------------------+----------+-------------+-------------+-----------------------
 grigorijschleifer | grigorijschleifer | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
 postgres             | postgres          | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
 template             | postgres          | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres+postgres=CTc/postgres
 template1            | postgres          | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres +postgres=CTc/postgres
(4 rows)
```
No mimic there;

I tried to connect via psql command but get the following error:

```code
~ psql 'dbname=mimic user=mimicuser options=--search_path=mimiciii'
psql: FATAL:  role "mimicuser" does not exist 
```
or: 

```code
grigorijschleifer=# \connect mimiciii
FATAL:  database "mimiciii" does not exist
Previous connection kept
```
In PostgreSQL 11 I can see only three databases:

![image](https://user-images.githubusercontent.com/36699154/91886040-03675600-ec89-11ea-8d7d-7cfc3e938200.png)

I also get <strong>OperationalError: FATAL:  database "mimic" does not exist</strong> when trying a connection from the <a href="https://github.com/MIT-LCP/mimic-code/blob/master/notebooks/ipynb_example/icu_length_of_stay.ipynb">icu_length_of_stay.ipynb</a> notebook.
 
One issue that could have caused this is maybe my uninstall and reinstall of homebrew a couple of months ago. Since then, many things behave spooky ...

Any advice is more than welcome!

Thank you for your time in advance :smile:

**Comment by alistairewj** [comment]
You might have installed MIMIC with a different database name. We use the
database name 'mimic' as a convention, but the data could equally we'll be
installed in one of your other three databases: grigorijschleifer,
postgres, or template0. I'd navigate into those databases and look for the
MIMIC-III schema or the tables. By the sound of the hard drive space you
have the data just in an unorthodox location!

>

**Comment by GrigorijSchleifer** [comment]
Hi Alistaire, thank you for the feedback. Unfortunately, I am not able to detect it at all. I know the database is there, because of the disk space usage. I followed your advice and connected to all visible databases, but they are all small and do not contain mimic information. 

Do you know the "usual" location where mimic is installed on a Mac machine or any postgres command that can scan for hidden databases?

Thank you!

**Comment by alistairewj** [comment]
Yeah actually this page is quite useful: https://wiki.postgresql.org/wiki/Disk_Usage

That should give you the queries you need to find out what tables in the DB are taking up all the space.

**Comment by GrigorijSchleifer** [comment]
Hi Alistaire, thank you for your help! Hope you are well.

**Comment by GrigorijSchleifer** [comment]
Interestingly, the terminal command 

`sudo find / -size +500000 -print`

finds some big files and they could belong to mimic.

/usr/local/var/postgres/base/17112/17429
/usr/local/var/postgres/base/17112/17445
/usr/local/var/postgres/base/17112/17421
/usr/local/var/postgres/base/17112/17417
/usr/local/var/postgres/base/17112/17547.1
/usr/local/var/postgres/base/17112/17409.1
....

**Comment by alistairewj** [comment]
There's a bunch of reasons to not be able to connect - in order I'd check (1) the postgres service is running and (2) you are connecting to the database as the `postgres` DB user.

Once you can connect try this query:

```sql
SELECT d.datname as Name,  pg_catalog.pg_get_userbyid(d.datdba) as Owner,
    CASE WHEN pg_catalog.has_database_privilege(d.datname, 'CONNECT')
        THEN pg_catalog.pg_size_pretty(pg_catalog.pg_database_size(d.datname))
        ELSE 'No Access'
    END as Size
    , tb.spcname as tablespace_name
    , pg_tablespace_location(tb.oid) AS tablespace_loc
FROM pg_catalog.pg_database d
LEFT JOIN pg_catalog.pg_tablespace tb
  on d.dattablespace = tb.oid
    order by
    CASE WHEN pg_catalog.has_database_privilege(d.datname, 'CONNECT')
        THEN pg_catalog.pg_database_size(d.datname)
        ELSE NULL
    END desc -- nulls first
    LIMIT 20;
```

That will tell you where the data reside. If it's blank it's all in the default PGDATA location which is shown by `SHOW data_directory;`.

**Comment by GrigorijSchleifer** [comment]
Hi Alistaire, thank you again for your help. I've checked and the service is running. 

`postgresql@11 started grigorijschleifer/Users/grigorijschleifer/Library/LaunchAgents/homebrew.mxcl.postgresql@11.plist` 

I had to install Postgresql version 11. The reason for that was that I found the following error message in the log file.

`The data directory was initialized by PostgreSQL version 11, which is not compatible with this version 13.0.`

I just realized that I only have myself as a DB user.

![image](https://user-images.githubusercontent.com/36699154/102698988-c1ffb400-4241-11eb-8376-7d6f9a919464.png)

On a second PC, where I am able to connect to MIMIC I can see the `postgres` user is initiated.

Because I deleted the postgres folder in order to free space on my machine, I will reinstall the DB and see if I can get it to run again with the postgres user installed.

Thank you!

Greetings from Lockdown-Germany ... Stay safe!
